ToDo/Program.cs

```diff
@@ -1,200 +1,191 @@
-            return menuInfo.ToString();
-        }
-
-        public static void ShowTaskList()
-        {
-            List<Task> taskList = _listOfTasks.GetTaskList();
-
-            for (int i = 0; i < taskList.Count(); i++)
-            {
-                if (_positionInTask == i)
-                {
-                    Console.ForegroundColor = ConsoleColor.Blue;
-                }
-                else
-                {
-                    Console.ForegroundColor = ConsoleColor.Yellow;
-                }
-
-                var builder = new StringBuilder();
-                builder
-                    .Append($"{i + 1}")
-                    .AppendLine($"\tName: {taskList[i].Title}")
-                    .AppendLine($"\tDescription: {taskList[i].Description}")
-                    .AppendLine($"\tCreated At: {taskList[i].CreatedAt}")
-                    .AppendLine($"\tCompleted At: {(taskList[i].IsDone ? taskList[i].CompletedAt : "Not complite")}");
-
-                Console.WriteLine(builder.ToString());
-                if (!_listOfTasks.IsEmpty())
-                {
-                    Console.ForegroundColor = ConsoleColor.Green;
-                    Console.WriteLine(new string('-', 40));
-                }
-            }
-        }
-
-        public static int GetNumberOfTask()
-        {
-            return _listOfTasks.GetTaskList().Count;
-        }
-
-        public static bool IsValid(string userInput)
-        {
-
-            return userInput != string.Empty;
-        }
-
-        public static bool IsIndexValid(int index, string wayToMove)
-        {
-            bool moveUP = 0 < index;
-            bool moveDown = index < _numberOfTask - 1;
-
-            if (wayToMove == "up")
-            {
-                return moveUP;
-            }
-            else
-            {
-                return moveDown;
-            }
-        }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ToDo
+{
+    class Program
+    {
+        private static TaskList _listOfTasks = new();
+        private static int _positionInTask = 0;
+        private static int _numberOfTask;
+        private static Menu _menu = new("Main Menu");
+
+        static void Main(string[] args)
+        {
+            _menu.AddNewItem("add new task", "a", ExecuteAddNewTask);
+            _menu.AddNewItem("add description", "e", ExecuteAddDescription);
+            _menu.AddNewItem("remove task", "x", ExecuteDeleteTask);
+            _menu.AddNewItem("mark task as done", "d", () => ExecuteStatusTask("done"));
+            _menu.AddNewItem("mark task as undone", "u", () => ExecuteStatusTask("undone"));
+            _menu.AddNewItem("move up", "w", () => ExecuteChangePosition(_positionInTask, "up"));
+            _menu.AddNewItem("move down", "s", () => ExecuteChangePosition(_positionInTask, "down"));
+
+            while (true)
+            {
+                Console.WriteLine(NavigationMenu());

```

Comment:
Move "Console.WriteLine" to "NavigationMenu". Rename NamigationMenu to smth like "PrintNavigationMenu". Pass the _menu as parameter to PrintNavigationMenu